### PR TITLE
fix: check end of stream of sound

### PR DESCRIPTION
**Pull request by pathway27**
Wondering if this is what's required for https://github.com/defold/defold/issues/5270 ?

### CODE REVIEW

File: engine/sound/src/sound.cpp [hunk 1]
```diff
@@ -788,7 +788,7 @@ namespace dmSound
 
     bool IsPlaying(HSoundInstance sound_instance)
     {
-        return sound_instance->m_Playing; // && !sound_instance->m_EndOfStream;
+        return sound_instance->m_Playing && !sound_instance->m_EndOfStream;
```
Comment (JCash) [inline engine/sound/src/sound.cpp:791]: Have you run the tests?
The first thing that happened is that it failed.

Also, when running the repro case with this fix, the app crashed.
Did you test your fix?

I don't know _why_ it was commented out, but never removed, but it's slightly more complicated than that. You need to read the code, test it, and reason about why your fix works (and put your reasoning in the PR)

**Review by JCash — CHANGES_REQUESTED** [review]

**Comment by pathway27** [comment]
Sorry something came up, I will let someone else take over.

**Comment by marcopar** [comment]
looks like this is closed but not fixed?
shouldn't be opened again?

**Comment by britzl** [comment]
@marcopar I think it's better if the PR remains closed. If someone else want to attempt a fix for #5270 they should open a new PR.

**Comment by marcopar** [comment]
Oh sure... I thought this was the bug report.... 
:facepalm:

**Pull request closed without merging**